src/rules/rules.js

```diff
@@ -203,45 +221,86 @@ function setEnabled (uid, isEnabled) {
   if (!_ruleExists(uid)) {
     throw Error('There is no rule with UID ' + uid);
   }
-  RuleManager.setEnabled(uid, isEnabled);
+  ruleManager.setEnabled(uid, isEnabled);
 }
 
 /**
-  * Creates a rule. The rule will be created and immediately available.
-  *
-  * @example
-  * import { rules, triggers } = require('openhab');
-  *
-  * rules.JSRule({
-  *  name: "my_new_rule",
-  *  description: "this rule swizzles the swallows",
-  *  triggers: triggers.GenericCronTrigger("0 30 16 * * ? *"),
-  *  execute: (event) => { // do stuff }
-  * });
-  *
-  * @memberof rules
-  * @param {RuleConfig} ruleConfig The rule config describing the rule
-  * @returns {HostRule} the created rule
-  * @throws {Error} an error if the rule with the passed in uid already exists and {@link RuleConfig.overwrite} is not `true`
-  */
-function JSRule (ruleConfig) {
-  const ruleUID = ruleConfig.id?.replace(/\W/g, '-') || ruleConfig.name.replace(/\W/g, '-') + '-' + randomUUID();
-  if (ruleConfig.overwrite === true) {
-    removeRule(ruleUID);
+ * Creates a {@link https://github.com/openhab/openhab-core/blob/main/bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/RuleImpl.java org.openhab.core.automation.internal.RuleImpl}.
+ *
+ * <p>This function allows creating rules that execute JavaScript code in a sandboxed context that is not shared with the script that created the rule.
+ * This does NOT allow using anything defined outside the rule's callback function, such as functions, classes or variables from the context of the script that created the rule.
+ * This approach, however, has the benefit that the rule runs in its own context and can therefore run at any time without the need to wait for other rule executions to complete.
+ *
+ * <p>Through the {@link RuleConfig} it can be opted into creating these rules instead of SimpleRules by the {@link rules.JSRule} function.
+ *
+ * @param {string} ruleUID
+ * @param {RuleConfig} ruleConfig
+ * @return {HostRule} the created {@link https://github.com/openhab/openhab-core/blob/main/bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/RuleImpl.java RuleImpl}
+ * @private
+ */
+function _createRule (ruleUID, ruleConfig) {
+  let script = ruleConfig.execute.toString();
+  script = script.match(/(function)?[^{]+\{([\s\S]*)}$/);
```

Comment:
So, this is where "the magic" happens? I don't quite get that you can extract the source code just by calling `toString()` on the function, but it's great that you can.